packages/shell-api/src/database.ts

```diff
@@ -149,10 +150,25 @@ export default class Database extends ShellApiWithMongoClass {
   }
 
   // Private helpers to avoid sending telemetry events for internal calls. Public so rs/sh can use them
-
   public async _runCommand(
     cmd: Document,
     options: CommandOperationOptions = {}
+  ): Promise<Document> {
+    return this._mongo._serviceProvider.runCommandWithCheck(
+      this._name,
+      adjustRunCommand(cmd, this._instanceState.shellBson),
+      {
+        readPreference: 'primaryPreferred',
```

Comment:
I would imagine that we don’t want to apply any read preference here at all, to match what drivers and (presumably) the legacy shell do?

Reply:
OK, will fix everything in the same commit as the previous left over.